packages/@webex/common/src/browser-detection.js

```diff
@@ -44,8 +44,22 @@ const createDetectionObject = (results) => {
   };
 };
 
-export default memoize((agent) =>
-  agent || window.navigator?.userAgent
-    ? createDetectionObject(bowser.getParser(agent || window.navigator.userAgent))
-    : mockDetectionObject
-);
+let browserDetectionData;
+
+const checkBrowserDetection = (agent) => {
+  if (browserDetectionData) {
+    return browserDetectionData;
+  }
+
+  if (!agent || !window.navigator?.userAgent) {
+    return mockDetectionObject;
+  }
+
+  const browserData = createDetectionObject(bowser.getParser(agent || window.navigator.userAgent));
```

Comment:
We should perform a quick check here specifically for the `browser name` to validate it exists on the `browserData` object. If not, we should return the mock detection data again. This should cover the linked task as intended (just for browser name)

Reply:
Added blow